### PR TITLE
Switch default routing/layout method to sabre for opt level 3

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit switches the default routing and layout method for
optimization level 3 to use `SabreSwap` and `SabreLayout` passes. The
quality of results is typically better with sabre than the default
stochastic swap and dense layout we're using now. For optimization
level 3 where we try to produce the best quality result and runtime
is of secondary concern using sabre makes the most sense (especially
after #7036 which improves the runtime performance). Also for
optimization level 3 currently sabre is typically faster because we
increase the number of trials for stochastic swap (which is generally
significantly faster) which slows it down as it's doing more work.
This should improve the quality and speed of the results in general
when running with optimization level 3. In the future we can do more
work to improve the runtime performance of the sabre passes and
hopefully make it fast enough to use for all the optimization levels
which have more constraints on the run time performance than level 3.

### Details and comments

Related to #7112 and #7200

### CODE REVIEW

**Comment by mtreinish** [comment]
Marking as on hold until #7036 merges because I don't think we want to do this without having that performance improvement

**Comment by mtreinish** [comment]
To see the improvement in the quality of results you can look at the nightly benchmarks here: 

- https://qiskit.github.io/qiskit/#transpiler_qualitative.TranspilerQualitativeBench.track_depth_transpile_mod8_10_178?p-optimization%20level=3&x-axis=routing%20method&p-layout%20method='dense'&p-layout%20method='sabre'&ram=64GB
- https://qiskit.github.io/qiskit/#transpiler_qualitative.TranspilerQualitativeBench.track_depth_transpile_4mod5_v0_19?ram=64GB&p-optimization%20level=3&p-routing%20method='sabre'&p-layout%20method='dense'&p-layout%20method='sabre'&x-axis=routing%20method
- https://qiskit.github.io/qiskit/#transpiler_qualitative.TranspilerQualitativeBench.track_depth_transpile_4gt10_v1_81?x-axis=routing%20method&python=3.8&p-optimization%20level=3&p-layout%20method='dense'&p-layout%20method='sabre'

Which is comparing the depth of the output circuits at optimization level 3 with different routing and layout method combinations.

**Comment by mtreinish** [comment]
Also: https://nonhermitian.org/posts/2021/2021-10-31-best_swap_mapper_qiskit.html is relevant here too

File: releasenotes/notes/sabre-opt-lvl3-default-550924470d683112.yaml
Comment (jakelishman) [inline releasenotes/notes/sabre-opt-lvl3-default-550924470d683112.yaml:18]:
```suggestion
    The default routing pass and layout pass for optimization level 3 has
    changed to use :class:`~qiskit.transpiler.passes.SabreSwap` and
    :class:`~qiskit.transpiler.passes.SabreLayout` respectively. This
    was done to improve the quality of the output result, as using the sabre
    passes produces better results than using
    :class:`~qiskit.transpiler.passes.StochasticSwap` and
    :class:`~qiskit.transpiler.passes.DenseLayout`, which were used as the
    defaults in prior releases. This change will improve the quality of the
    results when running :func:`~qiskit.compiler.transpile` or
    :func:`~qiskit.execute_function.execute` functions with the
    ``optimization_level`` kwarg set to ``3``. While this is generally an
    improvement, if you need to retain the previous behavior for any reason
    you can do this by explicitly setting the ``routing_method="stochastic"``
    and ``layout_method="dense"`` when calling
    :func:`~qiskit.compiler.transpile` with ``optimization_level=3``.
```

**Review by jakelishman — APPROVED** [review]
This is just a straightforwards swap - doesn't look like anything should go wrong, since we're already testing Sabre significantly.  I'll look at #7036 as well.

**Pull request merged**
